src/v/kafka/server/tests/alter_config_test.cc

```diff
@@ -847,3 +849,40 @@ FIXTURE_TEST(test_incremental_alter_config_remove, alter_config_test_fixture) {
     assert_property_value(
       test_tp, "iceberg.enabled", "false", new_describe_resp);
 }
+
+FIXTURE_TEST(test_alter_config_internal_topic, alter_config_test_fixture) {
+    wait_for_controller_leadership().get();
+    // create an internal topic
+    BOOST_REQUIRE(kafka::try_create_consumer_group_topic(
+                    app.coordinator_ntp_mapper.local(),
+                    app.controller->get_topics_frontend().local(),
+                    1)
+                    .get());
+    // enable authorization on it, to be able to make alter requests.
+    scoped_config config;
+    config.get("kafka_nodelete_topics").set_value(std::vector<ss::sstring>{});
+
+    absl::flat_hash_map<ss::sstring, ss::sstring> properties;
+    properties.emplace("icebert.enabled", "true");
+
+    auto resp = alter_configs(make_alter_topic_config_resource_cv(
+      model::kafka_consumer_offsets_topic, properties));
+    BOOST_REQUIRE_EQUAL(resp.data.responses.size(), 1);
+    BOOST_REQUIRE_EQUAL(
+      resp.data.responses[0].error_code, kafka::error_code::invalid_config);
+
+    absl::flat_hash_map<
+      ss::sstring,
+      std::pair<std::optional<ss::sstring>, kafka::config_resource_operation>>
+      incr_properties;
+    incr_properties.emplace(
+      "iceberg.enabled",
```

Comment:
Not necessarily related to this PR, but the a recent ask from product is to prefix this with "redpanda.", .e.g `redpanda.iceberg.enabled`

Reply:
oh I didn't know about this, the doc is stale looks like.